kalpsdk/write_transaction.go

```diff
@@ -8,6 +8,67 @@ import (
 	res "github.com/p2eengineering/kalp-sdk-public/response"
 )
 
+// Initialize initializes the smart contract owner by setting the owner ID.
+// It checks if there's already an owner, and if not, it sets the caller's userID as the owner ID.
+//
+// Returns:
+//   - error: An error if the operation fails.
+func (ctx *TransactionContext) Initialize() error {
+	ownerId, err := ctx.GetStub().GetState("smartContractOwner")
+	if err != nil {
+		return err
+	}
+
+	if string(ownerId) != "" {
+		return nil
+	}
+
+	userId, err := ctx.GetUserID()
+	if err != nil {
+		return err
+	}
+
+	err = ctx.GetStub().PutState("smartContractOwner", []byte(userId))
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// TransferOwner transfers ownership of the smart contract to the caller.
+// It verifies KYC completion for the new owner before transferring ownership.
+//
+// Returns:
+//   - error: An error if the operation fails.
+func (ctx *TransactionContext) TransferOwner(newUserId string) error {
+	isOwner, err := ctx.IsSmartContractOwner()
+	if err != nil {
+		return err
+	}
+	if !isOwner {
+		return fmt.Errorf("signer is not an owner")
+	}
+
+	// check Kyc for new owner userId
+	kycCheck, err := ctx.GetKYC(newUserId)
+	if err != nil {
+		return fmt.Errorf("failed to perform KYC check for user %s. Error: %v", newUserId, err)
+	}
+
+	// Return an error if the user has not completed KYC.
+	if !kycCheck {
+		return fmt.Errorf("user %s has not completed KYC", newUserId)
+	}
+
+	err = ctx.GetStub().PutState("smartContractOwner", []byte(newUserId))
```

Comment:
same hardcoded value. Defining const variable will be helpful if we need to udpate the const value. we can change it one place no need to copy paste at all the places.

Reply:
done